src/MinimalDelegation.sol

```diff
@@ -183,18 +184,45 @@ contract MinimalDelegation is
     }
 
     /// @inheritdoc ERC1271
+    /// @dev WrappedSignature is used for both NestedTypedDataSign and NestedPersonalSign signatures.
+    ///      If the caller implementing ERC-1271 is considered safe, we verify the signature over the data directly
+    /// TypedDataSign signatures are of the form: abi.encode(bytes32, bytes(`signature ‖ APP_DOMAIN_SEPARATOR ‖ contentsHash ‖ contentsDescr ‖ uint16(contentsDescr.length)`))
+    /// No extra data is needed for NestedPersonalSign signatures.
     function isValidSignature(bytes32 data, bytes calldata wrappedSignature)
         public
         view
         override(ERC1271, IERC1271)
         returns (bytes4 result)
     {
+        // Per ERC-7739, return 0x77390001 for the sentinel hash value
+        unchecked {
+            if (wrappedSignature.length == uint256(0)) {
+                // Forces the compiler to optimize for smaller bytecode size.
+                if (uint256(data) == ~wrappedSignature.length / 0xffff * 0x7739) return 0x77390001;
+            }
+        }
+
         (bytes32 keyHash, bytes memory signature, bytes memory hookData) =
             abi.decode(wrappedSignature, (bytes32, bytes, bytes));
-        bytes32 digest = _hashTypedData(data.hashWithWrappedType());
 
         Key memory key = getKey(keyHash);
-        bool isValid = key.verify(digest, signature);
+
+        bool isValid;
+        // Must be branched because we do abi decoding in memory which will throw since the encoding schemes are different
+        // ECDSA signatures are 65 bytes while P256 signatures are 64 bytes
+        if (signature.length == 64 || signature.length == 65) {
```

Comment:
Im confused by this branching, couldn't the signature be 64 or 65 in length AND still be a personal sign?

Reply:
hmmm, my understanding is that personal sigs are only called offchain... so it should never trigger the safe 1271 caller path

Reply:
and doesn't this codepath allow for personal sigs of length 64 | 65?